core/services/keeper/validate.go

```diff
@@ -33,6 +33,19 @@ decode_check_upkeep_tx   [type=ethabidecode
 encode_perform_upkeep_tx [type=ethabiencode
                           abi="performUpkeep(uint256 id, bytes calldata performData)"
                           data="{\"id\": $(jobSpec.upkeepID),\"performData\":$(decode_check_upkeep_tx.performData)}"]
+check_perform_upkeep_tx  [type=ethcall
+                          failEarly=true
```

Comment:
I think remove this flag. If the performUpkeep eth_call reverts, then something is wrong (it should never revert) and we want to know about it rather than fail silently.